pkg/tsdb/elasticsearch/response_parser.go

```diff
@@ -94,6 +103,8 @@ func parseResponse(ctx context.Context, responses []*es.SearchResponse, targets
 			logger.Debug("Processed metric query response")
 			if err != nil {
 				mt, _ := json.Marshal(target)
+				span.RecordError(err)
+				span.SetStatus(codes.Error, err.Error())
 				logger.Error("Error processing buckets", "error", err, "query", string(mt), "aggregationsLength", len(res.Aggregations))
 				return &backend.QueryDataResponse{}, err
 			}
```

Comment:
That's tricky, because we kind of need to record the error in both spans and also end both spans. I think for the outer span you can just do a `defer span.End()` after you created at, can't you?

Reply:
> you can just do a defer span.End() after you created it, can't you?

yes.

> That's tricky, because we kind of need to record the error in both spans and also end both spans. I

I guess we need to do 

```
span.RecordError(err)
span.SetStatus(codes.Error, err.Error())
resSpan.RecordError(err)
resSpan.SetStatus(codes.Error, err.Error())
resSpan.End()
```

and have `defer span.End()` so it will end as well. 

right?

Reply:
Yep, that sounds like a good plan.

Reply:
https://github.com/grafana/grafana/pull/74750/commits/2a322e3de38e78187f0e2c4013207e42ae35b28c